langchain/src/retrievers/parent_document.ts

```diff
@@ -197,16 +207,7 @@ export class ParentDocumentRetriever extends MultiVectorRetriever {
             metadata: { ...subDoc.metadata, [this.idKey]: parentDocId },
           })
       );
-      embeddedDocs.push(...taggedSubDocs);
-      fullDocs[parentDocId] = parentDoc;
-    }
-    if (this.childDocumentRetriever) {
-      await this.childDocumentRetriever.addDocuments(embeddedDocs);
-    } else {
-      await this.vectorstore.addDocuments(embeddedDocs);
-    }
-    if (addToDocstore) {
-      await this.docstore.mset(Object.entries(fullDocs));
+      await this._storeDocuments(parentDoc, taggedSubDocs, addToDocstore);
```

Comment:
Flagging that this does make it a bit more likely that we'll get into half-finished states